Game/RainMeadow.GameplayHooks.cs

```diff
@@ -251,6 +245,14 @@ private void ShelterDoorOnClose(On.ShelterDoor.orig_Close orig, ShelterDoor self
                 if (!realizedScug.readyForWin) return;
             }
             orig(self);
+            if (!self.IsClosing) return;
+
+            if (storyGameMode != null)
+            {
+                var storyClientSettings = storyGameMode.clientSettings as StoryClientSettings;
+                storyClientSettings.myLastDenPos = self.room.abstractRoom.name;
+                storyClientSettings.hasSheltered = true;
+            }
```

Comment:
I'll always frown a little bit at early-returns outside of a single-responsibility method. Makes mistakes harder to catch. 

Need another couple lines on the same hook? Didn't spot the early return? Blam.

Reply:
hm, I decided to use the early return because I don't think we'd want to do anything if the shelter is broken

Reply:
If we really want to check if the door is closing without using an early return, could we just wrap 249 and below with:

`if (self.IsClosing)`?

Reply:
yup, done